sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -307,6 +303,151 @@ export class ApplicationService extends RecordDataModelService<Application> {
     return { application, createdAssessment: originalAssessment };
   }
 
+  /**
+   * Starts a change request for an existing student application.
+   * A new application will be created in edited status and will be assessed by the Ministry.
+   * @param applicationId application ID to be changed.
+   * @param studentId student ID.
+   * @param applicationData application dynamic data to be saved.
+   * @param associatedFiles files to be associated with the application.
+   * @param auditUserId user that should be considered the one that is making the change request.
+   * @returns application ID of the created application that represents the change request.
+   */
+  async submitApplicationChangeRequest(
+    applicationId: number,
+    studentId: number,
+    applicationData: ApplicationData,
+    associatedFiles: string[],
+    auditUserId: number,
+  ): Promise<ApplicationSubmissionResult> {
+    const application = await this.getApplicationToSave(
+      studentId,
+      ApplicationStatus.Completed,
+      applicationId,
+    );
+    // If the application was not found it is because it does not belong to the student or it is not completed.
+    if (!application) {
+      throw new CustomNamedError(
+        "Student application not found or it is not in the correct status to be changed.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+    if (!allowApplicationChangeRequest(application.programYear)) {
+      throw new CustomNamedError(
+        "The program year associated to this application does not allow a change request submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+    // Check if there is already a change request in progress.
+    const hasChangeInProgress = application.parentApplication.versions.some(
+      (version) =>
+        [
+          ApplicationEditStatus.ChangeInProgress,
+          ApplicationEditStatus.ChangePendingApproval,
+        ].includes(version.applicationEditStatus),
+    );
+    if (hasChangeInProgress) {
+      throw new CustomNamedError(
+        "An application change request is already in progress.",
+        APPLICATION_CHANGE_REQUEST_ALREADY_IN_PROGRESS,
+      );
+    }
+    // Validates selected location and selected offering as key values that
+    // should not be changed. Even if some other values are changed (tampered by the user),
+    // there will be no impact for application processing since the location and offering
+    // are copied from the current application retrieved from the database.
+    if (
+      applicationData.selectedLocation !== application.data.selectedLocation
+    ) {
+      throw new CustomNamedError(
+        "Change request has a different location from its original submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+    if (
+      applicationData.selectedOffering !== application.data.selectedOffering
```

Comment:
What happens if the application submitted previously is requested for PIR? `undefined === undefined ` and it works?

Reply:
The expectation is that it must work. So far, the direction is that the user will not be able to change the location/offering, which means the application previously submitted as PIR will continue to be a PIR.